### PR TITLE
test free download and authticket download

### PR DESCRIPTION
**Pull request by Kishan-Dhakan**

### CODE REVIEW

File: tests/cli_tests/0_free_read_test.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	})
+
+	t.RunSequentially("Free reads should work", func(t *test.SystemTest) {
+		output, err := createWallet(t, configPath)
+		require.Nil(t, err, "Failed to create wallet", strings.Join(output, "\n"))
+
+		_ = setupWallet(t, configPath)
+
+		// Create an allocation
+		options := map[string]interface{}{"size": 1 * MB, "lock": "0.5"}
```
Comment (dabasov) [inline tests/cli_tests/0_free_read_test.go:70]: can we add readPrice here?

Comment (Kishan-Dhakan) [reply tests/cli_tests/0_free_read_test.go:70]: Is it needed? All blobbers have 0 read price now so default range is fine

Comment (dabasov) [reply tests/cli_tests/0_free_read_test.go:70]: if price changes to blobbers eventually, we can have an error early, but not have a false positive test with not free read

File: tests/cli_tests/0_free_read_test.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+	// revert read prices irrespective of test results
+	t.Cleanup(func() {
+		for _, blobber := range blobberList {
+			output, err := updateBlobberInfo(t, configPath, createParams(map[string]interface{}{"blobber_id": blobber.ID, "read_price": intToZCN(blobber.Terms.Read_price)}))
+			require.Nil(t, err, strings.Join(output, "\n"))
+		}
+	})
+
+	t.RunSequentially("Free reads should work", func(t *test.SystemTest) {
```
Comment (stewartie4) [inline tests/cli_tests/0_free_read_test.go:63]: Out of interest why is this sequential but the other test parallel?

File: tests/cli_tests/0_free_read_test.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+		}), true)
+		require.Nil(t, err, strings.Join(output, "\n"))
+		require.Len(t, output, 2)
+
+		require.Contains(t, output[1], StatusCompletedCB)
+		require.Contains(t, output[1], filepath.Base(filename))
+
+		downloadedFileChecksum := generateChecksum(t, "tmp/"+filepath.Base(filename))
+
+		require.Equal(t, originalFileChecksum, downloadedFileChecksum)
```
Comment (stewartie4) [inline tests/cli_tests/0_free_read_test.go:166]: nice

**Review by stewartie4 — APPROVED** [review]
Great coverage

**Comment by stewartie4** [comment]
tests not running because it looks like the "free-reads" branch doesn't exist in blobber repo

**Review by dabasov — APPROVED** [review]
lgtm

File: tests/cli_tests/0_free_read_test.go (hunk 2 above)
Comment (Kishan-Dhakan) [reply tests/cli_tests/0_free_read_test.go:63]: Nice catch, both should be parallel

**Pull request merged**